lib/feature/home_page/widgets/pokemon_search_page.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:pokedex_async_redux/api/models/pokemon_model.dart';
+import 'package:pokedex_async_redux/feature/home_page/widgets/pokemon_tile.dart';
+
+class PokemonSearchPage extends SearchDelegate {
+  PokemonSearchPage({required this.pokemonList});
+  final List<Pokemon> pokemonList;
+  @override
+  List<Widget>? buildActions(BuildContext context) => [
```

Comment:
the X icon, should only clear the search. not close the search. That's the job of the back arrow on the top left.